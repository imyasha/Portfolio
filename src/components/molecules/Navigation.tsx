import * as React from 'react';
import styled from 'styled-components';
import { motion } from 'framer-motion';
import { MenuItem } from 'Atoms/MenuItem';

const variants = {
  open: {
    pointerEvents: 'auto',
    transition: { staggerChildren: 0.07, delayChildren: 0.2 },
  },
  closed: {
    pointerEvents: 'none',
    transition: { staggerChildren: 0.05, staggerDirection: -1 },
  },
};

export const Navigation = ({ toggle }: { toggle: () => void }) => (
  <Menu variants={variants}>
    {itemIds.map((item) => (
      <MenuItem item={item} key={item.id} toggle={toggle} />
    ))}
  </Menu>
);

const itemIds = [
  {
    id: 0,
    title: 'Home',
    url: '/',
  },
  {
    id: 1,
    title: 'About',
    url: '/about',
  },
  {
    id: 2,
    title: 'Portfolio',
    url: '/portfolio',
  },
  {
    id: 3,
    title: 'Contact',
    url: '/contact',
  },
];

const Menu = styled(motion.ul).attrs(() => ({
  initial: 'closed',
  variants,
}))`
  margin: 0;
  padding: 0;
  padding: 40px;
  position: absolute;
  top: 100px;
  width: 230px;
`;
