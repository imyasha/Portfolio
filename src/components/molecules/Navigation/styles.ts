import styled from 'styled-components';

const StyledNavigation = styled.nav``;

const List = styled.ul`
	display: flex;
	align-items: center;
	flex-direction: column;
	font-size: 2rem;
	line-height: 2;
	font-weight: bold;

	.desktop-only {
		display: none;
	}

	& > li a {
		padding: 0.5rem 1rem;
		color: ${({ theme }) => theme.text};
		text-decoration: none;
		transition: 0.2s;

		&:hover {
			color: ${({ theme }) => theme.headline};
		}
	}

	@media screen and (min-width: 768px) {
		flex-direction: row;
		font-size: 1rem;
		font-weight: normal;

		.desktop-only {
			display: block;
		}
	}
`;

export { StyledNavigation, List };
