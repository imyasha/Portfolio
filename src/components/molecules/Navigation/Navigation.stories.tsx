import { Navigation } from './Navigation';

export default {
	title: 'molecules/Navigation',
	component: Navigation,
};

export const NavigationStory = () => <Navigation />;

NavigationStory.storyName = 'Navigation';
