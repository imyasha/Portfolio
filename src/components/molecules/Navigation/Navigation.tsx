import { FC } from 'react';
import Link from 'next/link';
import { List, StyledNavigation } from './styles';

import { ThemeToggle } from 'Atoms/ThemeToggle';

interface NavigationProps {
	onClick?: () => void;
}

const Navigation: FC<NavigationProps> = ({ onClick }) => {
	return (
		<StyledNavigation>
			<List>
				<li>
					<Link href="/">
						<a onClick={onClick}>Home</a>
					</Link>
				</li>
				<li>
					<Link href="/about">
						<a onClick={onClick}>About</a>
					</Link>
				</li>
				<li>
					<Link href="/uses">
						<a onClick={onClick}>Uses</a>
					</Link>
				</li>
				<li>
					<Link href="/blog">
						<a onClick={onClick}>Blog</a>
					</Link>
				</li>
				<li>
					<Link href="/portfolio">
						<a onClick={onClick}>Portfolio</a>
					</Link>
				</li>
				<li className="desktop-only">
					<ThemeToggle />
				</li>
			</List>
		</StyledNavigation>
	);
};

export { Navigation };
