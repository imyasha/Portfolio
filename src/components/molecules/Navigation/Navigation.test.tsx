import { Navigation } from '../Navigation';
import { render } from '@testing-library/react';

const mockDispatch = jest.fn();

jest.mock('react-redux', () => ({
	useSelector: jest.fn(),
	useDispatch: () => mockDispatch,
}));

describe('Navigation', () => {
	it('renders correctly', () => {
		const { container } = render(<Navigation />);
		expect(container).toMatchSnapshot();
	});
});
