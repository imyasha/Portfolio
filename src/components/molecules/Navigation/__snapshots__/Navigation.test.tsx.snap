// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation renders correctly 1`] = `
<div>
  <nav
    class="styles__StyledNavigation-sc-16epmp-0"
  >
    <ul
      class="styles__List-sc-16epmp-1 gPSCMv"
    >
      <li>
        <a
          href="/"
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/about"
        >
          About
        </a>
      </li>
      <li>
        <a
          href="/uses"
        >
          Uses
        </a>
      </li>
      <li>
        <a
          href="/blog"
        >
          Blog
        </a>
      </li>
      <li>
        <a
          href="/portfolio"
        >
          Portfolio
        </a>
      </li>
      <li
        class="desktop-only"
      >
        <button
          class="styles__StyledThemeToggle-sc-1dfmkyd-0 iOJryI"
          title="Switch to light theme"
        >
          <test-file-stub />
        </button>
      </li>
    </ul>
  </nav>
</div>
`;
